x-pack/plugins/fleet/server/services/agent_policy_watch.ts

```diff
@@ -99,7 +99,7 @@ export class PolicyWatcher {
         policiesToUpdate.map((policy) => {
           const { id, revision, ...policyContent } = policy;
           return {
-            type: AGENT_POLICY_SAVED_OBJECT_TYPE,
+            type: LEGACY_AGENT_POLICY_SAVED_OBJECT_TYPE,
```

Comment:
Seeing a lot of these hard-coded SO types referencing legacy. Is there a point at which any of these will need to switch over to the new SO type if/when we migrate?

I can see we are _writing_ to the new index as soon as the user opts-in but do we need to read from both too? Will these legacy not be stale during/after migration?

Reply:
Yes we should switch to the new SO type after the migration, for now I focused on fixing the agent_policy service but all the other access to the SO should be dynamic too, and I will do the switch as part of that PR

I do not think we need to read from both, we will need to read either from the LEGACY before the migration or from the new one after.